tsl/src/continuous_aggs/options.c

```diff
@@ -284,3 +286,26 @@ continuous_agg_update_options(ContinuousAgg *agg, WithClauseResult *with_clause_
 		elog(ERROR, "cannot alter finalized option for continuous aggregates");
 	}
 }
+
+void
+warn_if_hierarchical_realtime_cagg(int32 parent_mat_hypertable_id, bool materialized_only)
+{
+	if (materialized_only)
+		return;
+
+	ContinuousAgg *parent_cagg;
+	int32 mat_hypertable_id = parent_mat_hypertable_id;
+
+	while ((parent_cagg = ts_continuous_agg_find_by_mat_hypertable_id(mat_hypertable_id)) !=
+		   NULL)
+	{
+		if (!parent_cagg->data.materialized_only)
+		{
+			elog(NOTICE,
```

Comment:
another option could be to use ereport and put `consider using materialized_only=true` inside errhint, but I don't know if that would be better, just a suggestion.

Reply:
looks better - thank you for the suggestion!